lib/helpers/botConfig.js

```diff
@@ -15,56 +15,85 @@ function fetchByCampaignId(campaignId) {
 }
 
 /**
+ * @param {Object} entry
+ * @param {String} templateName
+ * @return {Object}
+ */
+function getTemplateConfigFromEntryAndTemplateName(entry, templateName) {
+  const contentTypeName = contentful.parseContentTypeFromEntry(entry);
+  return config.templatesByContentType[contentTypeName][templateName];
+}
+
+/**
+ * @param {Object} entry
  * @param {String} templateName
  * @return {String}
  */
-function getDefaultTextForBotConfigTemplateName(templateName) {
-  return config.templates.botConfig[templateName].default;
+function getDefaultValueFromEntryAndTemplateName(entry, templateName) {
+  const templateConfig = module.exports
+    .getTemplateConfigFromEntryAndTemplateName(entry, templateName);
+  return templateConfig.default;
 }
 
 /**
- * @param {Object} botConfig
+ * @param {Object} entry
+ * @param {String} templateName
+ * @return {String}
+ */
+function getFieldValueFromEntryAndTemplateName(entry, templateName) {
+  const templateConfig = module.exports
+    .getTemplateConfigFromEntryAndTemplateName(entry, templateName);
+  const fieldName = templateConfig.fieldName;
+  return entry.fields[fieldName];
+}
+
+/**
+ * @param {Object} entry
  * @param {String} templateName
  * @return {Object}
  */
-function getTemplateFromBotConfigAndTemplateName(botConfig, templateName) {
-  const botConfigValue = this.getTemplateTextFromBotConfigAndTemplateName(botConfig, templateName);
-  if (botConfig && botConfigValue) {
+function getTemplateFromEntryAndTemplateName(entry, templateName) {
+  const fieldValue = module.exports.getFieldValueFromEntryAndTemplateName(entry, templateName);
+  if (fieldValue) {
     return {
-      raw: botConfigValue,
+      raw: fieldValue,
       override: true,
     };
   }
+
   return {
-    raw: this.getDefaultTextForBotConfigTemplateName(templateName),
+    raw: module.exports.getDefaultValueFromEntryAndTemplateName(entry, templateName),
     override: false,
   };
 }
 
 /**
- * @return {Array}
+ * @param {Object} entry
+ * @return {Object}
  */
-function getTemplatesFromBotConfig(botConfig) {
+function getTemplatesFromEntry(entry) {
```

Comment:
I was debating on using `FromContentfulEntry` instead of `Entry,`, would that help?  My hesitation for renaming to `FromConfig` is that it seems unclear whether its a contentful entry like a `botConfig`, `postTextConfig`, or `photoPostConfig` or a file in the root `config` dir.